File: backend/netlify/functions/helpers/supabase.js
const { createClient } = require('@supabase/supabase-js');

const supabaseURL = process.env.SUPABASE_URL;
const supabaseKey = process.env.SUPABASE_KEY;

if (!supabaseURL || !supabaseKey) {
    throw new Error('Missing supabase url or key.');
}

const supabase = createClient(supabaseURL, supabaseKey);

module.exports = supabase;

File: backend/utils/supabase.py
from supabase import create_client

supabase = create_client(SUPABASE_URL, SUPABASE_KEY)

def insert_recipe(recipe):
    """Insert a new recipe into Supabase"""
    response = supabase.table("recipes").insert(recipe).execute()
    if response.error:
        raise Exception(response.error.message)
    